packages/next/server/next-server.ts

```diff
@@ -235,6 +235,8 @@ export default class NextNodeServer extends BaseServer {
       process.env.__NEXT_SCRIPT_WORKERS = JSON.stringify(true)
     }
 
+    process.env.NEXT_RUNTIME = 'nodejs'
```

Comment:
@huozhi I did this like you did in https://github.com/vercel/next.js/blob/a720dbdc4c0411ff0384449d00e125b4d3f0f1c5/packages/next/bin/next.ts#L96 that should be fine right?

we need to do this because when the server is instanciated through here, `process.env.NEXT_RUNTIME` doesn't get inlined by webpack so we need to define it manually

Reply:
`typeof window` condition should work already, we don't need to change all the condition since swc is already doing that work. will check the `process.env.NEXT_RUNTIME` coverage then, kinda wired it doesn't work with test